movie_library/views/movie_controller.py

```diff
@@ -0,0 +1,73 @@
+from django.http import HttpResponse
+from rest_framework.viewsets import ViewSet
+from rest_framework.response import Response
+from rest_framework import status
+from ..components.movie_component import MovieComponent
+from ..models.movie import Movie
+from ..utils.serializers.movie_serializer import MovieSerializer
+from ..repositories.movie_repository import MovieRepository
+
+class MovieViewSet(ViewSet):
+
+    def list(self, request):
+
+        movies = MovieComponent.get_movies()
+        result = MovieSerializer(movies, many=True)
+        return Response(data=result.data, status=200)
+
+    def create(self, request):
+
+        request_data = request.data
+        flag = MovieComponent.is_movie_exist(request_data)
+
+        #try:
+        if flag == True:
+             #movie.save()
+             movie = MovieComponent.create_new_movie(request_data)
+             result = MovieSerializer(movie, many=False)
+             return Response(data=result.data, status=201)
+        #except:
+            #serializer = MovieSerializer(Movie(request_data))
+        else:
+          return Response('There is a movie with this title',status=status.HTTP_400_BAD_REQUEST)
+            #raise ValueError('No file found')
```

Comment:
ass better approach you should always add the return at the end of the function as possible as you can